### PR TITLE
Formatting -> Format

### PR DESCRIPTION
**Pull request by timholy**
Formatting.jl is now archived.

EDIT: This also adds [compat] bounds on stdlibs, and makes Julia 1.6 the minimum supported version.

### CODE REVIEW

**Comment by timholy** [comment]
Test failures expected from #73. Fortunately the PDB tests run before the Pfam tests.

**Comment by timholy** [comment]
Hmm, looks like we have to drop Julia 1.0 if we're going to do this. I've tentatively made Julia 1.6 the minimum. I've also done a minor bump, since you're supposed to when you drop support for older Julia versions.

**Comment by timholy** [comment]
Test failures are because Interpro throttles the number of simultaneous users. There may not be much reason to test unix/windows/mac, perhaps just one would suffice.

**Comment by diegozea** [comment]
Hi @timholy, thank you so much for tackling this; it was on my todo list for a while :)

Yes, making 1.6 the minimum Julia version makes sense, as it is the LTS version. I was trying to keep compatibility with older Julia versions, as the official version in my institute's cluster is 1.5. I think that old Julia versions could also be present in other academic HPC systems. However, developing such old Julia versions becomes complicated as the ecosystem evolves, so let's move on to 1.6 as the minimum.

Yes, testing the download in a single OS makes sense to avoid that problem.

File: src/PDB/PDBParser.jl [hunk 1]
```diff
@@ -150,7 +150,7 @@ const _Format_PDB_ATOM = FormatExpr(
     #          1         2         3         4         5         6         7         8
     # 12345678901234567890123456789012345678901234567890123456789012345678901234567890
     # >    <>   < >  <|> < |>  <|   >      <>      <>      <>    <>    <      >  <><><
-    "{:<6}{:>5d} {:<4}{:>1}{:>3} {:>1}{:>4}{:>1}   {:>8.3f}{:>8.3f}{:>8.3f}{:>6.2f}{:>6.2f}      {:<4}{:>2}{:>2}\n"
+    "{:<6}{:>5d} {:<4}{:>1}{:>3} {:>1}{:>4}{:>1}   {:>8.3f}{:>8.3f}{:>8.3f}{:>6.2f}{:>6}      {:<4}{:>2}{:>2}\n"
```
Comment (diegozea) [inline src/PDB/PDBParser.jl:153]: Hi @timholy , 

Is there a reason to have more than 2 decimal places for the B-factors? 

The [PDB format description](https://www.wwpdb.org/documentation/file-format-content/format23/sect9.html) describe this field as:
```
COLUMNS      DATA TYPE        FIELD      DEFINITION
------------------------------------------------------
61 - 66      Real(6.2)        tempFactor Temperature factor.
```

Comment (timholy) [reply src/PDB/PDBParser.jl:153]: Oh shoot, I don't know why I did that.

**Comment by timholy** [comment]
> as the official version in my institute's cluster is 1.5

My condolences 😆

**Pull request merged**